bundles/org.openhab.binding.miele/src/main/java/org/openhab/binding/miele/internal/discovery/MieleMDNSDiscoveryParticipant.java

```diff
@@ -70,83 +79,88 @@ public void modified(@Nullable Map<String, Object> configProperties) {
             }
         }
     }
 
     @Override
     public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
-        return Collections.singleton(MieleBindingConstants.THING_TYPE_XGW3000);
+        return MieleBridgeHandler.SUPPORTED_THING_TYPES;
     }
 
     @Override
     public String getServiceType() {
-        return "_mieleathome._tcp.local.";
+        return SERVICE_TYPE;
     }
 
     @Override
-    public DiscoveryResult createResult(ServiceInfo service) {
-        if (isMieleGateway(service)) {
-            ThingUID uid = getThingUID(service);
-
-            if (uid != null) {
-                Map<String, Object> properties = new HashMap<>(2);
-
-                InetAddress[] addresses = service.getInetAddresses();
-                if (addresses.length > 0 && addresses[0] != null) {
-                    properties.put(MieleBindingConstants.HOST, addresses[0].getHostAddress());
-
-                    Socket socket = null;
-                    try {
-                        socket = new Socket(addresses[0], 80);
-                        InetAddress ourAddress = socket.getLocalAddress();
-                        properties.put(MieleBindingConstants.INTERFACE, ourAddress.getHostAddress());
-                    } catch (IOException e) {
-                        logger.error("An exception occurred while connecting to the Miele Gateway : '{}'",
-                                e.getMessage());
-                    }
-                }
-
-                return DiscoveryResultBuilder.create(uid).withProperties(properties)
-                        .withRepresentationProperty(MieleBindingConstants.HOST).withLabel("Miele XGW3000").build();
-            }
+    public @Nullable DiscoveryResult createResult(ServiceInfo service) {
+        if (!isMieleGateway(service)) {
+            return null;
         }
-        return null;
+
+        ThingUID uid = getThingUID(service);
+        if (uid == null) {
+            return null;
+        }
+
+        InetAddress[] addresses = service.getInetAddresses();
+        if (addresses.length <= 0 || addresses[0] == null) {
+            return null;
+        }
+
+        String ipAddress = addresses[0].getHostAddress();
+        Map<String, Object> properties = new HashMap<>(2);
+        properties.put(MieleBindingConstants.HOST, ipAddress);
+
+        Socket socket = null;
+        try {
+            socket = new Socket(addresses[0], 80);
+            InetAddress ourAddress = socket.getLocalAddress();
+            String interfaceIpAddress = ourAddress.getHostAddress();
+            socket.close();
+
+            properties.put(MieleBindingConstants.INTERFACE, interfaceIpAddress);
+            logger.debug("Discovered Miele@home gateway with IP address {} and interface IP address {}", ipAddress,
+                    interfaceIpAddress);
+        } catch (IOException e) {
+            logger.error("An exception occurred while connecting to the Miele Gateway: '{}'", e.getMessage());
```

Comment:
warn is more adapted. error is reserved for critical problem impacting the full openHAB system.

Reply:
Fixed.